integration/feature/init/src/MillInitUtils.scala

```diff
@@ -0,0 +1,104 @@
+package mill.integration
+
+import mill.testkit.IntegrationTester
+import utest.*
+
+object MillInitUtils {
+  val defaultInitCommand =
+    Seq("init", "--base-module", "BaseModule", "--deps-object", "Deps", "--merge")
+
+  case class SplitResolvedTasks(
+      all: Seq[String],
+      successful: Seq[String],
+      failed: Seq[String]
+  ) {
+    val allSorted = all.sorted
+    {
+      val successfulAndFailedSorted = (successful ++ failed).sorted
+      require(allSorted == successfulAndFailedSorted, s"$allSorted != $successfulAndFailedSorted")
+    }
+  }
+  object SplitResolvedTasks {
+    def apply(
+        all: Seq[String] | Null = null,
+        successful: Seq[String] | Null = null,
+        failed: Seq[String] | Null = null
+    ) =
+      new SplitResolvedTasks(
+        if (all != null) all else successful ++ failed,
+        if (successful != null) successful else all diff failed,
+        if (failed != null) failed else all diff successful
+      )
+  }
+
+  /**
+   * @param expectedCompileTasks [[ None ]] to denote that the `resolve __.compile` task fails
+   * @param expectedTestTasks [[ None ]] to denote that the `resolve __.test` task fails
+   * @return
+   */
+  def testMillInit(
```

Comment:
If we're going to introduce this new helper, let's also use it in `MillInitMavenTests.scala` and `MillInitGradleTests.scala` to keep things consistent

Reply:
I have done this for the tests except for these 3 Maven tests, as their testing logic is quite different making assertions on whether thether `out` or `err` contain certain strings:

https://github.com/com-lihaoyi/mill/blob/8ba432279ac643d98e8484aeae3e92b8d7b83f6f/integration/feature/init/src/MillInitMavenTests.scala#L14-L133